### PR TITLE
feat: add cw-hub-bnUSD contract

### PR DESCRIPTION
**Pull request by qwerty0789**

### CODE REVIEW

File: contracts/token-contracts/cw-hub-bnusd/src/contract.rs [hunk 1]
```diff
@@ -331,6 +354,18 @@ mod execute {
             .add_attribute("method", "x_cross_transfer_revert")
             .add_event(event))
     }
+
+    fn hex_encode(data: Vec<u8>) -> String {
+        debug_println!("this is {:?}", data);
+        if data.is_empty() {
+            debug_println!("this is empty");
+            "null".to_string()
```
Comment (nightowl121) [inline contracts/token-contracts/cw-hub-bnusd/src/contract.rs:362]: You can encode null values as rlp encoding.

File: contracts/cw-common/src/network_address.rs [hunk 2]
```diff
@@ -134,14 +123,61 @@ impl FromStr for NetworkAddress {
...
 }
 
+fn validate_icon_address(address: &str) -> bool {
+    let lowercase_address = address.to_lowercase();
+
+    if !lowercase_address.starts_with("hx") && !lowercase_address.starts_with("cx") {
+        return lowercase_address.len() == 40 && is_valid_character_set(&lowercase_address);
+    }
+
+    lowercase_address.len() == 42 && is_valid_character_set(&lowercase_address[2..])
```
Comment (nightowl121) [inline contracts/cw-common/src/network_address.rs:133]: Shouldn't we check here for if address starts with hx or cx here?

File: contracts/token-contracts/cw-hub-bnusd/src/contract.rs (hunk 1 above)
Comment (qwerty0789) [reply contracts/token-contracts/cw-hub-bnusd/src/contract.rs:362]: This is done only for the event logs, so

File: contracts/cw-common/src/network_address.rs (hunk 2 above)
Comment (qwerty0789) [reply contracts/cw-common/src/network_address.rs]: It was checked, but I guess there was unnecessary checks inside that, so removed in new commit

File: contracts/token-contracts/cw-hub-bnusd/src/contract.rs (hunk 1 above)
Comment (nightowl121) [reply contracts/token-contracts/cw-hub-bnusd/src/contract.rs:362]: okay, sounds good.

**Pull request merged**